sklearn/mixture/gmm.py

```diff
@@ -448,14 +455,20 @@ def fit(self, X):
             log_likelihood = []
             # reset self.converged_ to False
             self.converged_ = False
+
+            # this line should be removed when 'thresh' is deprecated
+            tol = self.tol if self.thresh is None \
+                else self.thresh / float(X.shape[0])
+
             for i in range(self.n_iter):
                 # Expectation step
                 curr_log_likelihood, responsibilities = self.score_samples(X)
-                log_likelihood.append(curr_log_likelihood.sum())
+                log_likelihood.append(curr_log_likelihood.mean())
 
                 # Check for convergence.
+                # (should compare to self.tol when 'thresh' is deprecated)
                 if i > 0 and abs(log_likelihood[-1] - log_likelihood[-2]) < \
-                        self.thresh:
+                        tol:
```

Comment:
actually here, it would be more readable to introduce:

```python
log_likelihood_change = abs(log_likelihood[-1] - log_likelihood[-2])
if i > 0 and log_likelihood_change < tol:
    ...
```